ztp/siteconfig-generator/siteConfig/siteConfigBuilder.go

```diff
@@ -413,3 +451,92 @@ func (scbuilder *SiteConfigBuilder) splitYamls(yamls []byte) ([][]byte, error) {
 
 	return resources, nil
 }
+
+func (scbuilder *SiteConfigBuilder) generateDiskImageRegistry(clusterId int, siteConfigTemp SiteConfig) error {
+	cluster := siteConfigTemp.Spec.Clusters[clusterId]
+
+	for ndId, node := range cluster.Nodes {
+		fmt.Println(ndId)
+		fmt.Println(node)
+		var configM []byte
+		var disks []ign3types.Disk
+		var filesystems []ign3types.Filesystem
+		var units []ign3types.Unit
+		for _, dPart := range node.DiskPartition{
+			diskDevice :=  dPart.Device
+			// for each device create a partition
+			for _, partition := range dPart.Partitions {
+				partitionLable := partition.Label
+
+				diskSizeMiB := partition.Size
+				diskStartMiB := partition.Start
+				disk := ign3types.Disk{
+					Device: diskDevice,
+					Partitions: []ign3types.Partition{
+						{
+							Label:    &partitionLable,
+							SizeMiB:  &diskSizeMiB,
+							StartMiB: &diskStartMiB,
+						},
+					},
+				}
+				disks = append(disks, disk)
+
+				fileSystemDevice := "/dev/disk/by-partlabel/" + partitionLable
+				fileSystemLable := "/" + strings.ReplaceAll(partitionLable, "-", "/")
+				fileSystemFormat := "xfs"
```

Comment:
Should the filesystem be configurable (but default to xfs)?

Reply:
XFS is the default of RHEL 7 and over. But I forget...did we decide to make this configurable?

Reply:
Defaulting to xfs but is now configurable